src/helpers.ml

```diff
@@ -50,28 +50,67 @@ let trim_comments comment =
   in
   aux comment 0 false
 
-let github_repo_of_gitlab_project_path ~gitlab_mapping gitlab_full_name =
+let github_repo_of_gitlab_project_path ~gitlab_mapping ~gitlab_domain
+    ~gitlab_repo_full_name =
+  let full_name_with_domain = gitlab_domain ^ "/" ^ gitlab_repo_full_name in
   let github_full_name =
-    match Hashtbl.find gitlab_mapping gitlab_full_name with
+    match Hashtbl.find gitlab_mapping full_name_with_domain with
     | Some value ->
         value
     | None ->
         Stdio.printf
           "Warning: No correspondence found for GitLab repository %s.\n"
-          gitlab_full_name ;
-        gitlab_full_name
+          full_name_with_domain ;
+        gitlab_repo_full_name
   in
   match Str.split (Str.regexp "/") github_full_name with
   | [owner; repo] ->
       (owner, repo)
   | _ ->
-      failwith "Could not split github_full_name into (owner, repo)."
+      failwith
+        (f "Could not split repository full name %s into (owner, repo)."
+           github_full_name )
 
-let github_repo_of_gitlab_url ~gitlab_mapping gitlab_repo_url =
-  let owner, repo =
-    if not (string_match ~regexp:".*:\\(.*\\)/\\(.*\\).git" gitlab_repo_url)
-    then Stdio.printf "Could not match project name on repository url.\n" ;
-    (Str.matched_group 1 gitlab_repo_url, Str.matched_group 2 gitlab_repo_url)
+let parse_gitlab_repo_url ~http_repo_url =
+  if not (string_match ~regexp:"https?://\\([^/]*\\)/\\(.*/.*\\)" http_repo_url)
+  then failwith "Could not match project name on repository url.\n" ;
+  (Str.matched_group 1 http_repo_url, Str.matched_group 2 http_repo_url)
+
+let%expect_test "http_repo_url_parsing_coq" =
+  let gitlab_domain, gitlab_repo_full_name =
+    parse_gitlab_repo_url ~http_repo_url:"https://gitlab.com/coq/coq"
+  in
+  Stdio.print_endline gitlab_domain ;
+  Stdio.print_endline gitlab_repo_full_name ;
+  [%expect {|
+     gitlab.com
+     coq/coq |}]
```

Comment:
@Zimmi48 do you confirm that:

* the `let%expect_test … [%expect {|…|}]` perform some string trimming automatically?
  * if yes, maybe this test is not strict enough? (e.g., we wouldn't want to get `"coq/coq "` with a trailing space)
  * if not, I believe the test will fail?
* what is the command-line necessary to run all the tests? (admittedly, was lazy to double-check the doc :)
* I guess these new tests are not already run in a GHA CI test:
  * do you think it would be practical to add a GHA workflow within this PR, or after merging it? or before?
  * in the later case, may I suggest to use `coq-community/docker-coq-action@v1`… because it is definitely not Coq-specific, but opam-specific :)

Reply:
>the `let%expect_test … [%expect {|…|}]` perform some string trimming automatically?

Yes, on the expected output. I don't know if it trims the actual output.

>if yes, maybe this test is not strict enough? (e.g., we wouldn't want to get `"coq/coq "` with a trailing space)

Possibly. This should be tested.

>if not, I believe the test will fail?

Currently, it does not fail.

>what is the command-line necessary to run all the tests? (admittedly, was lazy to double-check the doc :)

`dune runtest` IIRC

>I guess these new tests are not already run in a GHA CI test:
>       
>* do you think it would be practical to add a GHA workflow within this PR, or after merging it? or before?
>* in the later case, may I suggest to use `coq-community/docker-coq-action@v1`… because it is definitely not Coq-specific, but opam-specific :)

Sure, we can add a GH Action. Currently, there is none that runs on PRs. Feel free to propose a setup in a separate PR.